doc/dev/debugging.rst

```diff
@@ -120,33 +120,8 @@ Pass Pipelines
 
 The compilation steps which take MLIR as an input and lower it to binary are broken into MLIR pass
 pipelines.  The ``pipelines`` argument of the ``qjit`` function may be used to alter the steps used
-for compilation. The default set of pipelines is defined via the ``catalyst.compiler.DEFAULT_PIPELINES``
-list. Its structure is shown below.
-
-.. code-block:: python
-
-    DEFAULT_PIPELINES = [
-        (
-            "HLOLoweringPass",
-            [
-                "canonicalize",
-                "func.func(chlo-legalize-to-hlo)",
-                "stablehlo-legalize-to-hlo",
-                "func.func(mhlo-legalize-control-flow)",
-                ...
-            ],
-        ),
-        (
-            "QuantumCompilationPass",
-            [
-                "lower-gradients",
-                "adjoint-lowering",
-                "convert-arraylist-to-memref",
-            ],
-        ),
-        ...
-        ]
-
+for compilation. The default set of pipelines is defined via the :func:`~.catalyst.compiler.get_stages`
+list.
```

Comment:
@erick-xanadu my main recommendation here would be that a reader at this point would likely be looking for the default set of pipelines information, and won't be able to find it (in the docs).

There are a couple of things we could do:

- Make `get_stages` show up in the docs, and extend the docstring to include the default pipeline.
  - Pro: docstring is very close to the code, someone updating the default pipeline would remember to update the docstring
  - Con: we don't typically have internal function docstrings visible in the catalyst docs.

- ?

Reply:
@josh146 Can you remind me here how to get functions to show up in the documentation? Thanks!

Reply:
This function doesn't currently show up in the docs because the docs only surfaces functions that are top level, in `catalyst.debug`, and in `catalyst.third_party`. 

I recommend aliasing this function in `catalyst.debug` (since this function is useful for debugging, and shows up in the `debugging.rst` guide!

Then, you just need to import this asliased function into the [`__init__.py`](https://github.com/PennyLaneAI/catalyst/blob/c0803e588c0abe50b6c4cc725b993c3b525dac87/frontend/catalyst/debug/__init__.py#L30), and add it to `__all__`.

Reply:
https://github.com/PennyLaneAI/catalyst/pull/1194/commits/dfc279984a390e407b141597e9deb68c6e12d8bf https://github.com/PennyLaneAI/catalyst/pull/1194/commits/46c47dfc64e0b259efab55ca02cfd5aec2c66f07

Reply:
https://xanaduai-pennylane--1194.com.readthedocs.build/projects/catalyst/en/1194/code/api/catalyst.debug.get_compilation_stages_groups.html